cryptography/src/lib.rs

```diff
@@ -131,12 +131,29 @@ pub trait BatchScheme: Specification {
 /// (which should be cheap to clone).
 pub trait Digest: Array + Copy {}
 
-/// An object that can generate a digest.
-pub trait Digestible<D: Digest>: Clone + Send + Sync + 'static {
-    /// Generate a digest from the object.
+/// An object that can be uniquely represented as a [Digest].
+pub trait Digestible<D: Digest>: Clone + Sized + Send + Sync + 'static {
+    /// Returns the unique representation of the object as a [Digest].
+    ///
+    /// If many objects can be represented by the same [Digest], you
```

Comment:
All other comments seem to point to `Digest` being completely `Unique`, whereas `Identity` is some sort of grouping.

This comment "If many objects can be represented by the same [Digest]..." is confusing because to use the types correctly, it should be the case that many objects should not be represented by the same digest.

I also wonder if the name of `identity()` could be clearer that it is not unique. Something like `trait Groupable` with `fn group_id()` (?)

Reply:
I agree we should assert uniqueness here. That's the typical understanding of a digest.